OpenRA.Mods.Common/Traits/Render/WithBuildingPlacedOverlay.cs

```diff
@@ -82,8 +84,11 @@ void INotifyDamageStateChanged.DamageStateChanged(Actor self, AttackInfo e)
 
 		void INotifyBuildingPlaced.BuildingPlaced(Actor self)
 		{
-			visible = true;
-			overlay.PlayThen(overlay.CurrentSequence.Name, () => visible = false);
+			if (!IsTraitDisabled)
+			{
+				visible = true;
```

Comment:
You'll need to override the trait disabled notification and set this to false, otherwise it will stay visible.